strapi/strapi
# chore: update to use react-router@6+

### What does it do?

* Updates react-router-dom to version 6
* Migrates our code so the app still works e.g. using a data-router, changing hooks to non-removed version
* adds deprecations for adding settings or menu `to` properties with absolute paths
* adds deprecations for passing async functions as the `Component` property

### Why is it needed?

* dependencies up to date!

### How to test it?

* The entire app is effected, the best plan would be a splat attack
* Try typical use cases, the E2E suite catches barely any of the app although this would have been the best way to ensure everything works.

### Related issue(s)/PR(s)

* resolves CONTENT-1827
* resolves CONTENT-1759
* resolves CONTENT-1758
* resolves CONTENT-1973
* resolves CONTENT-1757
* resolves CONTENT-1761
* resolves CONTENT-1763
* resolves CONTENT-2127
* resolves CONTENT-2128
* resolves CONTENT-2104
